src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/AzureSqlDatabaseBackupShortTermRetentionPolicyCmdletBase.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System.Collections.Generic;
+using Microsoft.Azure.Commands.Sql.Backup.Services;
+using Microsoft.Azure.Commands.Sql.Common;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
+using Microsoft.Azure.Commands.Sql.Database_Backup.Model;
+using Microsoft.Azure.Commands.Sql.Database.Model;
+
+namespace Microsoft.Azure.Commands.Sql.Database_Backup.Cmdlet
+{
+    public abstract class AzureSqlDatabaseBackupShortTermRetentionPolicyCmdletBase : AzureSqlCmdletBase<IEnumerable<AzureSqlDatabaseBackupShortTermRetentionPolicyModel>, AzureSqlDatabaseBackupAdapter>
+    {
+        /// <summary>
+        /// Parameter set with ResourceGroup name, Server name and Database name.
+        /// </summary>
+        protected const string PolicyByResourceServerDatabaseSet = "PolicyByResourceServerDatabase";
+
+        /// <summary>
+        /// Parameter set for using a Database Input Object.
+        /// </summary>
+        private const string PolicyByInputObjectSet = "PolicyByInputObject";
+
+        /// <summary>
+        /// Gets or sets the Database object to get the policy for.
+        /// </summary>
+        [Parameter(ParameterSetName = PolicyByInputObjectSet,
```

Comment:
I can't find parameter set for the case:
```
Get-AzureRmSqlDatabaseShortTermRetentionPolicy [-ResourceId] <resourceId>
```

Reply:
Is it necessary? I didn't think so because the resourceId will always be as follows:
/subscriptions/<subId>/resourceGroups/<rg>/providers/Microsoft.Sql/servers/<serverName>/databases/<dbName>/backupShortTermRetentionPolicies/default

Reply:
@dealaus all new cmdlets that perform operations on tracked resources should support the `ResourceId` parameter set to allow users to pipe the output of the generic resource cmdlets ([more information here](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/piping-in-powershell.md#using-the-resourceid-parameter))

The user should still be able to get the generic resource from the `Get-AzureRmResource` cmdlet and use it for this cmdlet; it shouldn't matter that the policy name is always "default" as the other components of the `resourceId` are variable

Reply:
Thanks for the explanation. The requirement makes much more sense now.